runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -84,6 +84,28 @@ computeServerMemoryState(TR::CompilationInfo *compInfo)
    return JITServer::ServerMemoryState::NORMAL;
    }
 
+/**
+ * @brief Helper method executed at the end of a compilation
+ * to determine whether the server is approaching the maximum number of active threads.
+ */
+JITServer::ServerActiveThreadsState
+computeServerActiveThreadsState(TR::CompilationInfo *compInfo)
+   {
+   int32_t highActiveThreadThreshold = TR::Options::getHighActiveThreadThreshold();
+   int32_t veryHighActiveThreadThreshold = TR::Options::getVeryHighActiveThreadThreshold();
+
+   // Typically getNumCompThreadsActive() needs to be protected by the compilationQueueMonitor,
+   // except here we allow some small imprecision because we implement a heuristic.
+   int32_t numOfActiveThreads = compInfo->getNumCompThreadsActive();
+
+   JITServer::ServerActiveThreadsState activeThreadState = numOfActiveThreads >= veryHighActiveThreadThreshold ?
```

Comment:
I think we should change `>=` to `>` to give us the option to disable this feature by providing thresholds equal to the maximum number of threads.